src/ensembl/src/shared/components/instant-download/instant-download-transcript/InstantDownloadTranscript.tsx

```diff
@@ -0,0 +1,215 @@
+import React, { useState, useEffect } from 'react';
+import classNames from 'classnames';
+import pick from 'lodash/pick';
+
+import { fetchForTranscript } from '../instant-download-fetch/fetchForTranscript';
+
+import InstantDownloadTranscriptVisualisation from './InstantDownloadTranscriptVisualisation';
+import Checkbox from 'src/shared/components/checkbox/Checkbox';
+import InstantDownloadButton from '../instant-download-button/InstantDownloadButton';
+
+import styles from './InstantDownloadTranscript.scss';
+
+type Layout = 'horizontal' | 'vertical';
+
+type TranscriptFields = {
+  id: string;
+  so_term: string;
+};
```

Comment:
Do we need to make the transcript type defined [here](https://github.com/Ensembl/ensembl-client/blob/dev/src/ensembl/src/content/app/entity-viewer/types/transcript.ts) a shared type and build these by picking the required properties from it?

Reply:
No, I don’t believe so. The types defined in the InstantDownload component describe the data that the component actually needs to make its decision. This data can be supplied by either passing to InstantDownload the whole Transcript object (thanks to TypeScript’s structural typing, where a parameter can have as many fields as it wants as long as it has the ones that the consumer requires), or just the few fields that InstantDownload needs.

Notice too that the Transcript type is defined in the context of EntityViewer; but InstantDownload will have to function both in EntityViewer and in GenomeBrowser.